app/src/components/features/rules/RuleBuilder/Header/ActionButtons/CreateRuleButton/actions/index.js

```diff
@@ -3,10 +3,103 @@ import { actions } from "../../../../../../../../store";
 //UTILS
 import { isValidUrl } from "../../../../../../../../utils/FormattingHelper";
 import { CONSTANTS as GLOBAL_CONSTANTS } from "@requestly/requestly-core";
-// LODASH
-import { inRange } from "lodash";
+import _, { inRange } from "lodash";
 import { getFunctions, httpsCallable } from "firebase/functions";
 import { ResponseRuleResourceType } from "types/rules";
+import {
+  extractInnerTextAndAttributesFromHTMLString,
+  getHTMLNodeName,
+  validateStringForSpecificHTMLNode,
+} from "./insertScriptValidators";
+import { isFeatureCompatible } from "utils/CompatibilityUtils";
+import FEATURES from "config/constants/sub/features";
+
+/**
+ * Convert user inputs into rule schema
+ *
+ * @param {*} ruleData
+ * @returns {
+ *  success: boolean,
+ *  newRuleData?: object, // in case of successful validation and transformation
+ *
+ *  // in case of failure
+ *  error?: string,
+ *  message?: string,
+ *  pairId?: string, // id of the pair that caused the error
+ * }
+ */
+export const transformAndValidateInputsforRule = async (ruleData) => {
+  switch (ruleData.ruleType) {
+    case GLOBAL_CONSTANTS.RULE_TYPES.SCRIPT: {
+      const isCompatibleWithAttributesForScriptRule = isFeatureCompatible(FEATURES.SCRIPT_RULE_HTML_BLOCK);
+
+      if (!isCompatibleWithAttributesForScriptRule) {
+        return ruleData;
+      }
```

Comment:
This should be moved into a separate function like `checkScriptRuleVersionCompatibility()`

```
function checkcheckScriptRuleVersionCompatibility() {
    // Check for Attributes support added in Feb 2024
    const isCompatibleForAttribues = isFeatureCompatible(FEATURES.SCRIPT_RULE_ATTRIBUTES_SUPPORT);
    
    if (!isCompatibleForAttribues) return false;

    // We can similarly check other compatible things in future

}
```

Reply:
doing this just for this check would go agains the current convention across the code where we just use `isFeatureCompatible(<feature>)` for all such similar checks

Reply:
but extracting that to a variable here does seem redundant so replaced that with the function call itself